packages/incremental-merkle-tree.sol/contracts/LazyMerkleTree.sol

```diff
@@ -82,9 +82,30 @@ library LazyMerkleTree {
     }
 
     function root(LazyTreeData storage self) public view returns (uint256) {
-        // this will always short circuit if self.numberOfLeaves == 0
         uint40 numberOfLeaves = self.numberOfLeaves;
-        if (numberOfLeaves == 0) return defaultZero(1);
+        // dynamically determine a depth
+        uint8 depth = 1;
+        while (uint8(2)**depth < numberOfLeaves) {
+            depth++;
+        }
```

Comment:
You can calculate this in O(1) I think: `Math.ceil(Math.log2(numberOfLeaves))` in JS.

Reply:
Ah yes, that was my original approach but I don't think there's a cheap way to implement that in solidity.

Reply:
Right!